Sources/BrowserServicesKit/Email/EmailKeychainManager.swift

```diff
@@ -51,6 +51,46 @@ extension EmailKeychainManager: EmailManagerStorage {
     public func deleteAll() {
         EmailKeychainManager.deleteAll()
     }
+
+    public func getWaitlistToken() -> String? {
+        EmailKeychainManager.getString(forField: .waitlistToken)
+    }
+
+    public func getWaitlistTimestamp() -> Int? {
+        guard let timestampString = EmailKeychainManager.getString(forField: .waitlistTimestamp) else { return nil }
+        return Int(timestampString)
+    }
+
+    public func getWaitlistInviteCode() -> String? {
+        EmailKeychainManager.getString(forField: .inviteCode)
+    }
+
+    public func store(waitlistToken: String) {
+        // Avoid saving a new token if one already exists.
+        guard getWaitlistToken() == nil else { return }
+
+        EmailKeychainManager.add(waitlistToken: waitlistToken)
+    }
+
+    public func store(waitlistTimestamp: Int) {
+        // Avoid saving a new timestamp if one already exists.
+        guard getWaitlistTimestamp() == nil else { return }
+
+        let timestampString = String(waitlistTimestamp)
+        EmailKeychainManager.add(waitlistTimestamp: timestampString)
+    }
+
+    public func store(inviteCode: String) {
+        EmailKeychainManager.add(inviteCode: inviteCode)
+    }
+
+    public func deleteInviteCode() {
+        EmailKeychainManager.deleteItem(forField: .inviteCode)
+    }
+
+    public func deleteWaitlistState() {
+        EmailKeychainManager.deleteWaitlistState()
+    }
 }
 
 private extension EmailKeychainManager {
```

Comment:
General comment on Keychain storage: due to nature of the Keychain (storing outside of app sandbox) entries are not removed upon app reinstall. This differs from Android implementation and leaves potential concerns about "clean state" of the new installation - this most likely should be addressed, probably either by: 
 - Moving from keychain to other secure storage within app container.
 - Pairing key with UUID generated for every new installation so we don't reuse old ones (but that would leave orphaned entries there and cleanup is needed in such case).
 - Cleaning up keychain when app is being (re)installed (we'd need to set a flag in UserDefaults for that - and hope that there is no access issues when cold-starting that would wipe out the keychain by accident)